media/src/App.svelte

```diff
@@ -924,148 +934,33 @@
-                                        {uuid}
-                                        bind:color
-                                        bind:shadeUp
-                                        bind:animation
-                                        on:animate={animateIfNotAnimating}
-                                        selected={selectedObject === uuid}
-                                        on:click={selectObject(uuid)}
-                                        on:keydown={altDown}
-                                        {gridStep}
-                                        {gridMax}
-                                    />
-                                {:else if kind === 'vector'}
-                                    <Vector
-                                        {scene}
-                                        {onRenderObject}
-                                        {onDestroyObject}
-                                        bind:shadeUp
-                                        render={requestFrameIfNotRequested}
-                                        {uuid}
-                                        onClose={() => {
-                                            objects = objects.filter(
-                                                (b) => b.uuid !== uuid
-                                            );
-                                        }}
-                                        {params}
-                                        bind:color
-                                        bind:animation
-                                        on:animate={animateIfNotAnimating}
-                                        selected={selectedObject === uuid}
-                                        on:click={selectObject(uuid)}
-                                        on:keydown={altDown}
-                                        {gridStep}
-                                    />
-                                {/if}
+                                <svelte:component
+                                    this={kindToComponent[kind]}
+                                    {scene}
+                                    {onRenderObject}
+                                    {onDestroyObject}
+                                    camera={currentCamera}
+                                    controls={currentControls}
+                                    render={requestFrameIfNotRequested}
+                                    {params}
+                                    onClose={() => {
+                                        objects = objects.filter(
+                                            (b) => b.uuid !== uuid
+                                        );
+                                    }}
+                                    bind:color
+                                    bind:shadeUp
+                                    bind:animation
+                                    {uuid}
+                                    {gridStep}
```

Comment:
Looks like we need to pass in `{gridMax}` here as well to make Vector Field work with this refactor.